odometry:
  scan_matching:
    icp_objective: PointToPlane
    max_correspondence_dist: 1.0
    knn_normal_estimation: 4
    max_n_iter: 35
  scan_processing:
    voxel_size: -1.0
    downsampling_ratio: 0.3
    scan_cropping:
      cropping_radius: 40.0
      min_z: -30.0
      max_z: 30.0
      cropper_type: "MaxRadius" #otions are Cylinder, MaxRadius, MinRadius
  
mapping:
  is_print_timing_information: true
  is_build_dense_map: false
  is_attempt_loop_closures: true
  dump_submaps_to_file_before_after_lc: false
  min_movement_between_mapping_steps: 0.0
  submaps_num_scan_overlap: 10
  scan_to_map_refinement:
    min_refinement_fitness: 0.7
    scan_matching:
      icp_objective: PointToPlane
      max_correspondence_dist: 0.8
      knn_normal_estimation: 5
      max_n_iter: 50
    scan_processing:
      voxel_size: 0.05
      downsampling_ratio: 0.15
      scan_cropping:
        cropping_radius: 40.0
        min_z: -30.0
        max_z: 30.0
        cropper_type: "MaxRadius" #otions are Cylinder, MaxRadius, MinRadius

  submaps:
    size: 20
    min_num_range_data: 10
    adjacency_based_revisiting_min_fitness: 0.4
    
  map_builder:
    scan_cropping:
      cropping_radius: 40.0
      min_z: -50.0
      max_z: 50.0
      cropper_type: "MaxRadius" #otions are Cylinder, MaxRadius, MinRadius
    map_voxel_size: 0.1
    voxelize_every_n_scans: 5
    space_carving:
      voxel_size: 0.2
      max_raytracing_length: 30.0
      truncation_distance: 0.3
      carve_space_every_n_sec: 1.0
      min_dot_product_with_normal: 0.3
      
  dense_map_builder:
    scan_cropping:
      cropping_radius: 15.0
      min_z: -50.0
      max_z: 50.0
      cropper_type: "MaxRadius" #otions are Cylinder, MaxRadius, MinRadius
    map_voxel_size: 0.0
    voxelize_every_n_scans: 5
    space_carving:
      voxel_size: 0.1
      max_raytracing_length: 25.0
      truncation_distance: 0.1
      carve_space_every_n_sec: 3.0
      min_dot_product_with_normal: 0.1

  place_recognition:
    feature_map_normal_estimation_radius: 1.0
    feature_voxel_size: 0.5
    feature_radius: 2.5
    feature_knn: 100
    normal_knn: 10
    ransac_num_iter: 10000000
    ransac_probability: 0.999
    ransac_model_size: 3
    ransac_max_correspondence_dist: 0.75
    ransac_correspondence_checker_distance: 0.75
    ransac_correspondence_checker_edge_length: 0.5
    ransac_min_corresondence_set_size: 25
    max_icp_correspondence_distance: 0.2
    min_icp_refinement_fitness: 0.6 #the more aliasing, the higher this should be
    dump_aligned_place_recognitions_to_file: true 
    consistancy_check:
      max_drift_roll: 70.0 #deg
      max_drift_pitch: 70.0 # deg
      max_drift_yaw: 70.0 #deg
    
    
    
  global_optimization:
    edge_prune_threshold: 0.2
    loop_closure_preference: 2.0
    max_correspondence_distance: 10.0
    reference_node: 0

local_map:
  voxel_size: 0.08
  cropping_radius: 10.0

mesher:
  is_compute_mesh: false
  voxel_size: 0.2
  knn_normal_estimation: 4
  strategy: Poisson # options: AlphaShape, Poisson, BallPivot
  alpha_shape_alpha: 0.5
  poisson_depth: 8
  poisson_min_density: -2.0
  poisson_scale: 0.8
  ball_pivot_radii: [0.5] 
  
visualization:
  assembled_map_voxel_size: 0.2
  submaps_voxel_size: 0.2
  visualize_every_n_msec: 300.0
  
  


  

    